File: GameClient/Framework/Assets/GameLogic/ToLua/Misc/LuaProfiler.cs
using System.Collections.Generic;
//using UnityEngine.Profiling;

public static class LuaProfiler
{
    public static List<string> list = new List<string>();

    public static void Clear()
    {
        list.Clear();
    }

    public static int GetID(string name)
    {
        int id = list.Count;
        list.Add(name);
        return id;
    }

    public static void BeginSample(int id)
    {
        string name = list[id];
        UnityEngine.Profiling.Profiler.BeginSample(name);
    }

    public static void EndSample()
    {
        UnityEngine.Profiling.Profiler.EndSample();
    }
}

File: GameClient/Framework/Assets/GameLogic/Lua/EmmyLuaAPI/CSharpAPI/LuaProfiler.lua
---@class LuaProfiler : Object
---@field public list List
LuaProfiler={ }
---@public
---@return void
function LuaProfiler.Clear() end
---@public
---@param name string
---@return number
function LuaProfiler.GetID(name) end
---@public
---@param id number
---@return void
function LuaProfiler.BeginSample(id) end
---@public
---@return void
function LuaProfiler.EndSample() end
